projects/dsp-ui/src/lib/action/services/sorting.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SortingService {
+
+    constructor() { }
+
+    /**
+     * reverses an array
+     */
+    reverseArray(value: Array<any>): Array<any> {
+        if (value instanceof Array) {
+            return value.slice().reverse();
+        }
+    }
+
+    /**
+     * compares value by value and sorts by alphabetical order
+     * optionally, you can have the array returned to you in reversed order by providing the 'reversed' parameter to the method call
+     */
+    sortByAlphabetical(value: Array<any>, args: string, reversed?: boolean): Array<any> {
+        if (value instanceof Array) {
+            value.sort((a: any, b: any) => {
```

Comment:
I would be in favour of not changing the original array (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort). Could you call `slice()`on the array and then work on this copy?

Reply:
refactored in fe5dbc8